src/Directory.Build.targets

```diff
@@ -6,4 +6,16 @@
     <Exclude>[GraphQL.StarWars]*,[GraphQL.Harness]*,[GraphQL.Dummy]*</Exclude>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(IsPackable)' == 'true'">
+    <PackageReadmeFile>README.md</PackageReadmeFile>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <WarningsNotAsErrors>CS8604</WarningsNotAsErrors>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+
+  <ItemGroup Condition="'$(IsPackable)' == 'true'">
```

Comment:
From https://docs.microsoft.com/en-us/visualstudio/msbuild/customize-your-build?view=vs-2022 : 
> Include items in .props files (conditioned on a property). All properties are considered before any item, so user-project property customizations get picked up, and this gives the user's project the opportunity to Remove or Update any item brought in by the import.

Reply:
What do you think? Move back into .props? 

I have read documentation you provided and I'm fine to accept these changes.

Reply:
So that's why I was experiencing the weirdness -- it processed all the property groups first then the item groups.  I missed that when I was reading the doc.

Sure, I'll move the item groups back to `Directory.Build.props`, leaving the property conditions in `Directory.Build.targets`, and then merge if it all builds okay.

Reply:
Actually, I'll just do a separate PR so we can both review.

Reply:
I find all these MSBuild rules rather non-obvious. I have a nuget metapackage with props/targets that is used in all of our projects at my work. I spent a bunch of time and strength to place all the properties and items I need so that they work correctly. And I'm still not sure if I did it well.